Core/src/IfcGeometryConverter/GeomUtils.h

```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
 
+				/**********************************************************************************************/
+				/*! \brief Calculates coordinates of the intersection point.
+				*
+				* \param[in] initialPointSeg1		Initial point of the first intersecting line.
+				* \param[in] terminalPointSeg1		Terminal point of the first intersecting line.
+				* \param[in] initialPointSeg2		Initial point of the second intersecting line.
+				* \param[in] terminalPointSeg2		Terminal point of the second intersecting line.
+				* \param[out] intersectionPoint		Coordinates of the intersection point between first line segment and second line segment.
+				*
+				* \return							Returns true, if lines are intersecting with each other. False otherwise.
+				*/
 				static bool GeomUtils::LineToLineIntersection(const carve::geom::vector<2>& initialPointSeg1,
 					const carve::geom::vector<2>& terminalPointSeg1,
 					const carve::geom::vector<2>& initialPointSeg2,
 					const carve::geom::vector<2>& terminalPointSeg2,
 					carve::geom::vector<2> & intersectionPoint);
 
+				/**********************************************************************************************/
+				/*! \brief Appedns vector of points (in 3D) to the curve.
+				*
+				* \param[in] points_vec				A vector of points. 
+				*
+				* \param[out] target_vec			A curve. 
+				*/
 				static void appendPointsToCurve(
-					const std::vector<carve::geom::vector<3> >& points_vec,
-					std::vector<carve::geom::vector<3> >& target_vec);
+					const std::vector<carve::geom::vector<3>>& points_vec,
+					std::vector<carve::geom::vector<3>>& target_vec);
+
+				/**********************************************************************************************/
+				/*! \brief Appedns vector of points (in 2D) to the curve.
+				*
+				* \param[in] points_vec				A vector of points.
+				*
+				* \param[out] target_vec			A curve.
+				*/
 				static void appendPointsToCurve(
-					const std::vector<carve::geom::vector<2> >& points_vec,
-					std::vector<carve::geom::vector<3> >& target_vec);
-
+					const std::vector<carve::geom::vector<2>>& points_vec,
+					std::vector<carve::geom::vector<3>>& target_vec);
+
+				/**********************************************************************************************/
+				/*! \brief Computes an inverse of the matrix.
+				*
+				* \param[in] matrix_a				An initial invertible matrix.
+				*
+				* \return							Inverted matrix
+				*/
 				static carve::math::Matrix computeInverse(
 					const carve::math::Matrix& matrix_a);
 
+				/**********************************************************************************************/
+				/*! \brief Calculates a closest point on the line to the point. 
+				*
+				* \param[in] point					Point outside the line. 
+				* \param[in] line_origin			Line origin point. 
+				* \param[in] line_direction			Direction of the line. 	
+				*
+				* \param[out] closest				Closest point on the line to the point.
+				*/

```

Comment:
I am not sure, if this description is correct. Can you please give me a feedback regarding this doxy-comment

Reply:
See my suggestion. I believe that you've described it quite nicely, only a few corrections from my side to make it even more clear.